app/controllers/clients_controller.rb

```diff
@@ -0,0 +1,80 @@
+class ClientsController < ApplicationController
+  before_action :set_client, only: [:show, :edit, :update, :destroy]
+
+  # GET /clients
+  # GET /clients.json
+  def index
+    @clients = Client.all
+  end
+
+  # GET /clients/1
+  # GET /clients/1.json
+  def show
+    @identities = Identity.where(client_id: params[:id])
+
+    if Movement.count > 0
+      @movements = Movement.where(client_id: params[:id])
+    end
+
+  end
+
+  # GET /clients/new
+  def new
+    @client = Client.new
+  end
+
+  # GET /clients/1/edit
+  def edit
+  end
+
+  # POST /clients
+  # POST /clients.json
+  def create
+    @client = Client.new(client_params)
+
+    respond_to do |format|
+      if @client.save
+        format.html { redirect_to @client, notice: 'Client was successfully created.' }
+        format.json { render :show, status: :created, location: @client }
+      else
+        format.html { render :new }
+        format.json { render json: @client.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /clients/1
+  # PATCH/PUT /clients/1.json
+  def update
+    respond_to do |format|
+      if @client.update(client_params)
```

Comment:
i don't think we need any of these -- we're only viewing data - not modifying it